admin/admin-filters-term.php

```diff
@@ -284,9 +286,13 @@ protected function save_language( $term_id, $taxonomy ) {
 
 			$language = $this->model->get_language( sanitize_key( $_POST['term_lang_choice'] ) );
 
-			if ( ! empty( $language ) ) {
-				$this->model->term->set_language( $term_id, $language );
+			if ( empty( $language ) ) {
+				return;
 			}
+
+			( new User() )->can_translate_or_die( $language );
+
+			$this->model->term->set_language( $term_id, $language );
```

Comment:
Almost everything in this method comes too late (which assigns language to newly created terms).